### ISSUE TITLE
[Bug] Backup with external S3 storage fails during upload phase with large backup file

### ISSUE THREAD

**Issue by gpantanetti** [mariadb-operator/mariadb-operator#900]
**Documentation**

- [x] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**

I have a *Backup* configuration with an external S3 storage that includes some small and large databases on a MariaDB Operator Galera Cluster.

The upload phase of the backup *Job* fails with this messages:

```
2024-10-02T13:25:29.243304427Z mariadb 💾 Exporting env
2024-10-02T13:25:29.256595104Z mariadb 💾 Writing target file: /backup/0-backup-target.txt
2024-10-02T13:25:29.259129463Z mariadb 💾 Taking backup: /backup/backup.2024-10-02T13:25:29Z.sql
2024-10-02T13:43:55.255149887Z mariadb-operator {"level":"info","ts":1727876635.2548766,"msg":"starting backup"}
2024-10-02T13:43:55.255386277Z mariadb-operator {"level":"info","ts":1727876635.255241,"msg":"configuring S3 backup storage"}
2024-10-02T13:43:55.277829542Z mariadb-operator {"level":"info","ts":1727876635.2776284,"msg":"reading target file","path":"/backup/0-backup-target.txt"}
2024-10-02T13:43:55.277866051Z mariadb-operator {"level":"info","ts":1727876635.2777286,"msg":"obtained target backup","file":"backup.2024-10-02T13:25:29Z.sql"}
2024-10-02T13:43:55.277874377Z mariadb-operator {"level":"info","ts":1727876635.2777438,"msg":"pushing target backup","file":"backup.2024-10-02T13:25:29Z.sql","prefix":"example/db"}
2024-10-02T13:45:19.621526840Z mariadb-operator {"level":"error","ts":1727876719.6210024,"msg":"error pushing target backup","file":"backup.2024-10-02T13:25:29Z.sql","prefix":"example/db","error":"Put \"https://*****/t85uyz6c57/example/db/backup.2024-10-02T13%3A25%3A29Z.sql?partNumber=4&uploadId=0ce8a21d-39ca-483b-8979-c8ca5fcf4c6c\": net/http: timeout awaiting response headers","stacktrace":"github.com/mariadb-operator/mariadb-operator/cmd/backup.init.func1\n\t/app/cmd/backup/main.go:89\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041\nmain.main\n\t/app/cmd/controller/main.go:415\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:272"}
```

In detail, the error occurring during the upload phase is:

```
Put "https://*****/t85uyz6c57/example/db/backup.2024-10-02T13%3A25%3A29Z.sql?partNumber=4&uploadId=0ce8a21d-39ca-483b-8979-c8ca5fcf4c6c": \
  net/http: timeout awaiting response headers
```

With this stacktrace:

```
github.com/mariadb-operator/mariadb-operator/cmd/backup.init.func1
	/app/cmd/backup/main.go:89
github.com/spf13/cobra.(*Command).execute
	/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989
github.com/spf13/cobra.(*Command).ExecuteC
	/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117
github.com/spf13/cobra.(*Command).Execute
	/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main
	/app/cmd/controller/main.go:415
runtime.main
	/usr/local/go/src/runtime/proc.go:272
```

**Note**: If the backup file is small enough - by excluding large databases from the *Backup* configuration - the upload phase is completed without problems and the backup file is correctly saved on the external S3 storage.

**Expected behaviour**

The *Backup* completes without errors and the backup file is saved on S3 external storage, even if the backup file is big.

**Steps to reproduce the bug**

1. Create some large database with MariaDB Operator
2. Configure a *Backup* resource for the databases with an external S3 storage
3. Apply the *Backup* resource
4. Wait for the *Backup* job to fail during upload phase

I don't know how large the backup file should be for the upload phase to fail, my backup file is ~5 GB (and the upload bandwidth is ~100 Mbps).

**Environment details**:

- Kubernetes version: v1.28.13
- Kubernetes distribution: RKE2 on Ubuntu 20.04.6
- mariadb-operator version: v0.0.31
- Install method: Helm

**Comment by bennyvar77** [comment]
Hi
Could this be related to #786 ?
Have you tried "stagingStorage" which is included in the operator version 0.35.0?
https://github.com/mariadb-operator/mariadb-operator/releases/tag/0.35.0

**Comment by gpantanetti** [comment]
I don't think this could be related to #786 because the dump operation completes without problems; it's the upload phase to the S3 storage that fails.

**Update**: I've enabled the compression (available from v.0.34.0) and changed the S3 target region (choosing one that is closer to my infrastructure): this seems to have bypassed the problem and the upload phase is now completing without issues (total data is now ~1GB).
